### PR TITLE
Implement connect gaps for 2d WebGL plots

### PR DESCRIPTION
**Pull request by mikolalysenko**
This PR implements connect gaps for 2D webgl plots.  It requires the following upgrades to gl-line2d:

https://github.com/gl-vis/gl-line2d/commit/42172ed11a895103c402d11596d7e100b1abfeca

This should solve the following issue:

https://github.com/plotly/phoenix-integration/issues/44

A new test case (`gl2d_connect_gaps`) was added to the dashboard to verify that the feature is working:

screen shot 2016-04-19 at 3 19 44 pm

CC @bpostlethwaite @etpinard

### CODE REVIEW

**Comment by bpostlethwaite** [comment]
cc/ @monfera

**Comment by etpinard** [comment]
@mikolalysenko Something broke the [`gl2d_14.json`](https://github.com/plotly/plotly.js/blob/master/test/image/mocks/gl2d_14.json):

image


that line should be on follow the marker points.

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "cibuild": "node tasks/cibundle.js",
     "watch": "node tasks/watch_plotly.js",
     "lint": "eslint . || true",
+    "lint-fix": "eslint . --fix",
```
Comment (etpinard) [inline package.json]: nicely done!

**Comment by mikolalysenko** [comment]
@etpinard  Looks like the xData and yData arrays aren't getting initialized correctly for that plot. I can take a look at it next time I get a day to work.

File: src/traces/scattergl/convert.js [hunk 2]
```diff
@@ -460,16 +465,29 @@ proto.updateFancy = function(options) {
...
-        this.lineOptions.positions = positions;
+        var linePositions = positions;
+        if(!options.connectgaps) {
+            var p = 0;
+            var x = this.xData;
+            var y = this.yData;
+            linePositions = new Float32Array(2 * x.length);
+
+            for(i=0; i<x.length; ++i) {
+                linePositions[p++] = x[i];
```
Comment (etpinard) [inline src/traces/scattergl/convert.js]: you might have to :recycle:  this [block](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L363-L368).

**Comment by mikolalysenko** [comment]
@etpinard just a quick ping:  I fixed up the problem with the axes computations on some plots.  Let me know if this looks good now.

**Comment by etpinard** [comment]
@mikolalysenko `gl2d_14.json` still isn't quite right:

image

The logs coordinates show up as `NaN` on hover.

**Comment by mikolalysenko** [comment]
Alright, taking another look

**Comment by mikolalysenko** [comment]
Alright, I think I got the picking issue fixed.  Let me know if there is anything else out of the ordinary here.

**Comment by etpinard** [comment]
@mikolalysenko Looks great. Thanks!

**Pull request merged**